### ISSUE TITLE
tf.keras.layers.UpSampling2D crashes(aborts) when size is large

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46914]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.keras.layers.UpSampling2D` crashes(aborts) when `size` is large
**Describe the expected behavior**
expect an exception message if the input unexpected instead of crash. 

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
import numpy as np
tf.keras.layers.UpSampling2D(size=1610637938, data_format='channels_first', interpolation='bilinear')(np.ones((5,1,1,1)))
~~~
Output:
~~~python
2021-02-04 04:44:48.936606: F tensorflow/core/framework/tensor_shape.cc:353] Check failed: 0 <= new_num_elements (0 vs. -5475971237085092396)
Aborted (core dumped)
~~~

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/d0a9213b5a7d785b1aa18665710c18e8/46929.ipynb). Thanks!

**Comment by saikumarchalla** [comment]
Colab crashes till in TF 2.6 Nightly as well. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/2904b0ac3ee60c44defe00f3f2f5e72d/untitled93.ipynb).Thanks!

**Comment by yongtang** [comment]
Added a PR #51497 for the fix.

**Comment by oawxkw** [comment]
I also observed the following API aliases or similar APIs can cause the same issue in older versions of tensorflow.
Users should be cautious when using them on both CPU and GPU up to tensorflow 2.6.0 (v2.6.0-rc2-32-g919f693420e).

- `tf.keras.layers.UpSampling2D`, `tf.compat.v1.keras.layers.UpSampling2D`
- `tf.keras.layers.UpSampling1D`, `tf.compat.v1.keras.layers.UpSampling1D`
- `tf.keras.layers.UpSampling3D`, `tf.compat.v1.keras.layers.UpSampling3D`

It seems to be fixed in tensorflow 2.6.1 (v2.6.0-101-g3aa40c3ce9d) and later versions.

<details>
  <summary>Code to reproduce the issue in older versions of tensorflow</summary>

- `tf.keras.layers.UpSampling1D`, `tf.compat.v1.keras.layers.UpSampling1D`

```python
import numpy as np
import tensorflow as tf
print(tf.version.GIT_VERSION, tf.version.VERSION, flush=True)
print(tf.config.list_physical_devices(), flush=True)


try:
    tf.keras.layers.UpSampling1D(size=1610637938)(np.ones((5, 1, 1)))
    # tf.compat.v1.keras.layers.UpSampling1D(size=1610637938)(np.ones((5, 1, 1)))
except Exception as e:
    print("Error:", str(e), flush=True)
print("Success!", flush=True)
```

- `tf.keras.layers.UpSampling3D`, `tf.compat.v1.keras.layers.UpSampling3D`

```python
import numpy as np
import tensorflow as tf
print(tf.version.GIT_VERSION, tf.version.VERSION, flush=True)
print(tf.config.list_physical_devices(), flush=True)


try:
    tf.keras.layers.UpSampling3D(size=1610637938, data_format='channels_first')(np.ones((5, 1, 1, 1, 1)))
    # tf.compat.v1.keras.layers.UpSampling3D(size=1610637938, data_format='channels_first')(np.ones((5, 1, 1, 1, 1)))
except Exception as e:
    print("Error:", str(e), flush=True)
print("Success!", flush=True)
```
</details>
